Tests/AblyChatTests/DefaultRoomPresenceTests.swift

```diff
@@ -0,0 +1,465 @@
+import Ably
+@testable import AblyChat
+import Testing
+
+struct DefaultRoomPresenceTests {
+    // MARK: CHA-PR1
+
+    // @spec CHA-PR1
+    @Test
+    func channelNameIsSetAsChatMessagesChannelName() async throws {
+        // Given
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages")
+        let featureChannel = MockFeatureChannel(channel: channel)
+
+        // When
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "mockClientId", logger: TestLogger())
+
+        // Then
+        #expect(defaultPresence.channel.name == "basketball::$chat::$chatMessages")
+    }
+
+    // MARK: CHA-PR3
+
+    // @spec CHA-PR3a
+    // @spec CHA-PR3e
+    @Test
+    func usersMayEnterPresence() async throws {
+        // Given
+        let realtimePresence = MockRealtimePresence(members: ["client1"].map { .init(clientId: $0) })
+        let channel = MockRealtimeChannel(name: "basketball::$chat::$chatMessages", mockPresence: realtimePresence)
+        let featureChannel = MockFeatureChannel(channel: channel, resultOfWaitToBeAblePerformPresenceOperations: .success(()))
+        let defaultPresence = await DefaultPresence(featureChannel: featureChannel, roomID: "basketball", clientID: "client2", logger: TestLogger())
+
+        // When
+        try await defaultPresence.enter(data: ["status": "Online"])
+
+        // Then
+        let presenceMembers = try await defaultPresence.get()
```

Comment:
I think that these tests should be written in terms of the expected interactions with the underlying ably-cocoa object (i.e. asserting which methods were called and with which arguments) instead of writing a fake implementation of the ably-cocoa object and then checking the resulting state after we've interacted with it.

Reply:
@lawrence-forooghian Please check if that's what you are looking for https://github.com/ably/ably-chat-swift/pull/239/commits/39b431365aa8d83d0bde38bf8355c7ba86a7d703

Reply:
Yep, that's the general idea. I think that the idea of using `Any` for all of the argument types might bite us later, especially since I'm making the interface to ably-cocoa presence more strongly typed in my work for #244 (see e.g. https://github.com/ably/ably-chat-swift/compare/main...use-single-interface-to-ably-cocoa#diff-d7906db3f5740726ac6528e863379414afdea89862aa7871da80455c731fa03eR17 and how it now accepts a `JSONValue?` instead of `Any?` for the `data`) argument, so some of what you've done here might need to be rethought soon. But we can figure out what to do about it when the time comes.

Reply:
Agree, let's address this when needed, it will be easy to change I guess.